### DISCUSSION TITLE
How do I publish container port for my service, when I do not use traefik?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [basecamp/kamal#877] [Q&A]
I am deploying a Java app, which bundles in an embeddd http server,  so I do not want traefik sitting in front of it, no need for load balancing, or deployment ceremonies. How do I achieve  the same effect with Kamal configuration to do something like:
```
docker run -p 8080:8080 ...
```

**Comment by hrstoyanov** [comment]
Looking at the docs for Roles I see this example:
```yml
options:
      memory: 2g
      cpus: 4
 ```
Are these documented better anywhere?

**Reply by jeremedia** [reply]
This [stops traefik from being published](https://kamal-deploy.org/docs/configuration/traefik/#disabling-publishing).
```
traefik:
  publish: false
```

**Reply by hrstoyanov** [reply]
Thanks, do you have any inside for the ports mapping?
